programs/client/Client.cpp

```diff
@@ -706,6 +706,38 @@ bool Client::processWithFuzzing(const String & full_query)
         return true;
     }
 
+    const auto & settings = global_context->getSettingsRef();
+    const Dialect & dialect = settings.dialect;
+    String old_dialect;
+    switch (dialect)
+    {
+        case DB::Dialect::kusto:
+            old_dialect = "kusto";
+            break;
+        case DB::Dialect::clickhouse:
+            old_dialect = "clickhouse";
+            break;
+        case DB::Dialect::prql:
+            old_dialect = "prql";
+            break;
+    }
+
+    if (auto *q = orig_ast->as<ASTSetQuery>())
+    {
+        auto *setDialect = q->changes.tryGet("dialect");
+        if (setDialect)
+        {
+            old_dialect =  setDialect->get<String>();
+        }
+    }
+
+    //setting dialect to clickhouse during query fuzzing, restore dialect to original value after fuzzing
+    SettingChange new_setting("dialect", "clickhouse");
+    SCOPE_EXIT_SAFE({
+            global_context->setSetting("dialect", old_dialect);
+    });
+    global_context->applySettingChange(new_setting);
+
```

Comment:
do we need to switch back and forth if it's already "clickhouse"?

Reply:
fixed